code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/file/docker/OpendDockerComposeFile.xtend

```diff
@@ -0,0 +1,242 @@
+            )
+    }
+
+    /*
+     * Add or replace the docker compose parts based the the name of the intermediate operation
+     * node.
+     */
+    private def addOrReplaceToMap(String name, String dockerComposeServicePart) {
+        if (dockerComposeServiceParts.containsKey(name.toLowerCase)) {
+            dockerComposeServiceParts.replace(name.toLowerCase, dockerComposeServicePart)
+        } else {
+            dockerComposeServiceParts.put(name.toLowerCase, dockerComposeServicePart)
+        }
+    }
+
+    /*
+     * Create a POJO [Service] from  a container
+     */
+    private def String getServiceFromContainer(IntermediateContainer container) {
+        val service = new Service
+
+        service.build = Util::buildPathFromQualifiedName(
+            container.deployedServices.get(0).qualifiedName
+        )
+
+        service.image = container.name
+        service.container_name = container.name
+        container.endpoints
+
+        container.endpoints.forEach[endpoint |
+            endpoint.addresses.forEach[address |
+               service.ports.add(Util::getPortFromAddress(address) + ':'
+                    + Util::getPortFromAddress(address))
+            ]
+        ]
+
+        this.dockerComposeNetworkParts.forEach[networkPart |
+            service.networks.add(this.getNetworkName(networkPart))
+        ]
+
+        return DockerComposeTemplate::buildDockerComposeParts(container.name, service)
+    }
+
+    /*
+     * Create a POJO [Service] from  a infrastructure node
+     */
+    private def String getServiceFromInfrastructureNode(IntermediateInfrastructureNode node) {
+        val service = new Service
+        val imageTyp = node.defaultValues.findFirst[p |
+            p.technologySpecificProperty.name.toLowerCase == "imagetyp"
+            && p.value.toLowerCase == "springcomponent"
+        ]
+
+        if (imageTyp !== null) {
+            service.build = node.name + File.separator
+        }
+
+        service.image = node.name
+        service.container_name = node.name
+
+        node.endpoints.forEach[endpoint |
+            endpoint.addresses.forEach[address |
+                service.ports.add(Util::getPortFromAddress(address) + ':'
+                    + Util::getPortFromAddress(address))
+            ]
+        ]
+
+        this.dockerComposeNetworkParts.forEach[networkPart |
+            service.networks.add(this.getNetworkName(networkPart))
+        ]
+
+        node.dependsOnNodes.forEach[depentOnNode |
+            service.depends_on.add(depentOnNode.name)
+        ]
+
+        val stringBuilder = new StringBuilder
+        stringBuilder.append(DockerComposeTemplate::buildDockerComposeParts(node.name, service))
+
+        return stringBuilder.toString
+    }
+
+    /*
+     * Get the name for the network in the docker-compose file
+     */
+    private def String getNetworkName(String networkName) {
+        return networkName.lines.toArray.get(0)?.toString.replace(':', "").trim
```

Comment:
Why `toArray`? And why must `toString()` be called on the first element if `networkName` is already a `String`? In addition, I recognized that you mix characters and strings. At this occurrence, please use characters consistently to make sure the correct version of `replace` (i.e., the non-regex one) gets invoked:
```
replace(':', '')
```

On other occasions please use Strings, e.g.,
```
someStringVar + '\n'
```
becomes
```
someStringVar + "\n"
```

Or even better:

```
«someStringVar»\n
```


We rely on `Strings` whenever possible for consistency reasons.

Reply:
`'''«someStringVar»\n'''`
That is actually not working.

It should look like this:
`'''«line.toString.trim»«"\n"»'''`

Reply:
Ah, I see. You're obviously correct.

Other question: Where does actually the `lines` in `networkName.lines` come from? And furthermore: What type has the `line` variable? I still wonder about the call of `toString`.

Reply:
The lines for the network are extracted from an existing docker-compose file. The `toString` call is necessary to get the name of the network as a 'String' object.